### ISSUE TITLE
[Paper Note] DoReMi: Optimizing Data Mixtures Speeds Up Language Model Pretraining, Sang Michael Xie+, NeurIPS'23 Spotlight, 2023.05

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#698]
# URL
- http://arxiv.org/abs/2305.10429
# Authors
  - Sang Michael Xie
  - Hieu Pham
  - Xuanyi Dong
  - Nan Du
  - Hanxiao Liu
  - Yifeng Lu
  - Percy Liang
  - Quoc V. Le
  - Tengyu Ma
  - Adams Wei Yu
# Abstract
  - The mixture proportions of pretraining data domains (e.g., Wikipedia, books, web text) greatly affect language model (LM) performance. In this paper, we propose Domain Reweighting with Minimax Optimization (DoReMi), which first trains a small proxy model using group distributionally robust optimization (Group DRO) over domains to produce domain weights (mixture proportions) without knowledge of downstream tasks. We then resample a dataset with these domain weights and train a larger, full-sized model. In our experiments, we use DoReMi on a 280M-parameter proxy model to set the domain weights for training an 8B-parameter model (30x larger) more efficiently. On The Pile, DoReMi improves perplexity across all domains, even when it downweights a domain. DoReMi improves average few-shot downstream accuracy by 6.5% points over a baseline model trained using The Pile's default domain weights and reaches the baseline accuracy with 2.6x fewer training steps. On the GLaM dataset, DoReMi, which has no knowledge of downstream tasks, even matches the performance of using domain weights tuned on downstream tasks.
# Translation (by gpt-4o-mini)
- プレトレーニングデータドメインの混合比率（例：Wikipedia、書籍、ウェブテキスト）は、言語モデル（LM）の性能に大きく影響します。本論文では、最初にドメインに対するグループ分布頑健最適化（Group DRO）を用いて小さなプロキシモデルをトレーニングし、下流タスクの知識なしにドメインウェイト（混合比率）を生成する最適化手法であるドメイン再重み付け（DoReMi）を提案します。次に、これらのドメインウェイトを用いてデータセットを再サンプリングし、より大きなフルサイズモデルをトレーニングします。実験では、DoReMiを280Mパラメータのプロキシモデルに使用して、8Bパラメータのモデル（30倍の大きさ）をより効率的にトレーニングするためのドメインウェイトを設定しました。The Pileにおいて、DoReMiはすべてのドメインでパープレキシティを改善し、たとえドメインの重みを下げた場合でも効果があります。DoReMiは、The Pileのデフォルトのドメインウェイトを使用してトレーニングしたベースラインモデルに対して、平均的な少数ショット下流精度を6.5％ポイント向上させ、2.6倍少ないトレーニングステップでベースライン精度に達します。GLaMデータセットにおいても、DoReMiは下流タスクの知識がなくても、下流タスクでチューニングされたドメインウェイトを使用した際の性能と一致します。
# Summary (by gpt-4o-mini)
- ドメイン混合比率が言語モデルの性能に与える影響を考察し、Group DROを用いて小規模プロキシモデルをトレーニングし、ドメイン再重み付け（DoReMi）を提案。これにより、無知の下流タスクに基づきドメインウェイトを生成し、効率的に大規模モデルをトレーニング。DoReMiはThe Pileでパープレキシティを改善し、ベースラインモデルに対して少数ショット精度を6.5％向上させ、少ないトレーニングステップで結果を達成。GLaMデータセットでも同様の性能を示す。

**Comment by AkihikoWatanabe** [comment]
事前学習する際の各ドメインのデータをどのような比率でmixtureするかの話。各ドメインごとに小さなproxy modelを訓練し、downstream taskの知識無しでドメインごとの重みを生成。データセットを生成されたドメインごとの重みに従いリサンプリングすることで、（1/30のプロキシモデルを用いた場合）オリジナルのデータより2.6倍高速で、6.5%oneshotのaccuracyを向上させることに成功
![image](https://github.com/AkihikoWatanabe/paper_notes/assets/12249301/2c0b125a-5ecc-4ee3-8c3b-022c03606c60)

**Comment by AkihikoWatanabe** [comment]
openreview:https://openreview.net/forum?id=lXuByUeHhd
